core/eval.go

```diff
@@ -67,9 +67,14 @@ func evalSET(args []string) []byte {
 			return Encode(errors.New("ERR syntax error"), false)
 		}
 	}
+	valueInt, valueIntOrErr := strconv.ParseInt(value, 10, 64)
 
 	// putting the k and value in a Hash Table
-	Put(key, NewObj(value, exDurationMs, oType, oEnc))
+	if valueIntOrErr == nil {
+		Put(key, NewObj(valueInt, exDurationMs, oType, oEnc))
+	} else {
+		Put(key, NewObj(value, exDurationMs, oType, oEnc))
+	}
```

Comment:
Can we not check this while we are deducing the type and encoding? Doing `strconv` again is inefficient.

Reply:
Would have to make the deduce encoding method more specific for int64 usecases. It will have to return 4 params then instead of the existing two. One with the integer value and the other the err value of parseInt. And this will start bloating when more data types are added